lightning-invoice/src/utils.rs

```diff
@@ -66,28 +66,40 @@ pub fn create_phantom_invoice<Signer: Sign, K: Deref>(
 		invoice = invoice.amount_milli_satoshis(amt);
 	}
 
-	for PhantomRouteHints { channels, phantom_scid, real_node_pubkey } in phantom_route_hints {
-		let mut route_hints = filter_channels(channels, amt_msat);
-
-		// If we have any public channel, the route hints from `filter_channels` will be empty.
-		// In that case we create a RouteHint on which we will push a single hop with the phantom
-		// route into the invoice, and let the sender find the path to the `real_node_pubkey`
-		// node by looking at our public channels.
-		if route_hints.is_empty() {
-			route_hints.push(RouteHint(vec![]))
-		}
-		for mut route_hint in route_hints {
-			route_hint.0.push(RouteHintHop {
-				src_node_id: real_node_pubkey,
-				short_channel_id: phantom_scid,
-				fees: RoutingFees {
-					base_msat: 0,
-					proportional_millionths: 0,
-				},
-				cltv_expiry_delta: MIN_CLTV_EXPIRY_DELTA,
-				htlc_minimum_msat: None,
-				htlc_maximum_msat: None,});
-			invoice = invoice.private_route(route_hint.clone());
+	for hint in phantom_route_hints {
```

Comment:
Looks like somehow you ended up undoing some upstream changes? Not sure how that happened, but you'll probably need to manually `git reset` and commit.